### PR TITLE
[Samples] Add 11.qnamaker bot sample

### LINKED ISSUE #864: Sample: 11.qnamaker
Depends on #740

### PR DESCRIPTION
**Pull request by Batta32**
Fixes #864

## Description
We are adding the sample 11.qnamaker in Java taking into account the C# [11.qnamaker bot](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/csharp_dotnetcore/11.qnamaker) present in [BotBuilder-Samples](https://github.com/microsoft/BotBuilder-Samples) repository.

## Specific Changes
  - Generate main structure of the sample
  - Add `resources` folder with sample questions tsv
  - Add `deploymentTemplates` folder
  - Add root files
  - Add `application.properties`
  - Add LICENSE document
  - Add README document to describe the steps to run the sample

## Testing
_Current behavior of the Java migrated sample_
image

### CODE REVIEW

File: samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/QnABot.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        LoggerFactory.getLogger(QnABot.class).info("Calling QnA Maker");
+
+        QnAMakerOptions options = new QnAMakerOptions() {
+        	{
+        		setTop(1);
+        	}
+        };
+
+        // The actual call to the QnA Maker service.
+        qnaMaker.getAnswers(turnContext, options).thenAccept(response -> {
```
Comment (tracyboehrer) [inline samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/QnABot.java:44]: This should be:

```
        // The actual call to the QnA Maker service.
        return qnaMaker.getAnswers(turnContext, options)
            .thenCompose(response -> {
                if (response != null && response.length > 0) {
                    return turnContext.sendActivity(MessageFactory.text(response[0].getAnswer()))
                        .thenApply(sendResult -> null);
                }
                else {
                    return turnContext.sendActivity(MessageFactory.text("No QnA Maker answers were found."))
                        .thenApply(sendResult -> null);
                }
            });
```

File: samples/11.qnamaker/deploymentTemplates/new-rg-parameters.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (tracyboehrer) [inline samples/11.qnamaker/deploymentTemplates/new-rg-parameters.json:1]: new-rg-parameters.json can be removed.

File: samples/11.qnamaker/deploymentTemplates/preexisting-rg-parameters.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{
```
Comment (tracyboehrer) [inline samples/11.qnamaker/deploymentTemplates/preexisting-rg-parameters.json:1]: preexisting-rg-parameters.json can be removed.

File: samples/11.qnamaker/deploymentTemplates/template-with-new-rg.json [hunk 4]
```diff
@@ -0,0 +1,183 @@
+{
```
Comment (tracyboehrer) [inline samples/11.qnamaker/deploymentTemplates/template-with-new-rg.json:1]: Update template-with-new-rg.json with version from 02.echo-bot

File: samples/11.qnamaker/deploymentTemplates/template-with-preexisting-rg.json [hunk 5]
```diff
@@ -0,0 +1,154 @@
+{
```
Comment (tracyboehrer) [inline samples/11.qnamaker/deploymentTemplates/template-with-preexisting-rg.json:1]: Update template-with-preexisting-rg.json with version from 02.echo-bot

**Review by tracyboehrer — CHANGES_REQUESTED** [review]

**Comment by tracyboehrer** [comment]
@Batta32 Also need to update from main

**Comment by Batta32** [comment]
Thanks @tracyboehrer for your feedback. We will be applying the requested changes right now!

**Comment by Batta32** [comment]
@tracyboehrer - we already applied the requested changes and merged `main` into this branch 😊.

**Review by tracyboehrer — APPROVED** [review]

**Pull request merged**